compiler/AST/build.cpp

```diff
@@ -3130,3 +3133,15 @@ Expr* tryBangExpr(Expr* e)
 {
   return new CallExpr(PRIM_TRYBANG_EXPR, e);
 }
+
+Expr* convertAssignmentAndWarn(Expr* a, const char* op, Expr* b)
+{
+  if (0 == strcmp("=", op)) {
+    USR_FATAL_CONT(a, "Use == to check for equality in a conditional");
```

Comment:
I think the error message needs to be stronger, by saying "assignment is illegal in a conditional".
It, then, can go on to say what it says now.